### ISSUE TITLE
Alignment Tool: Basic MAP

### ISSUE THREAD

**Issue by klappy** [unfoldingWord/translationCore#2258]
![image](https://user-images.githubusercontent.com/13352518/39369031-cdd692a8-4a00-11e8-9d10-7fc4495a47c5.png)

- [x] MAP-suggested words show up with the dark green band
- [x] If the user clicks on the `x` the word returns to the wordbank
- [x] If the user clicks on the word, the band turns light blue and the `x` disappears
- [x] If the user clicks `Accept Suggestions`, all suggestions are suggested
- [x] If the user clicks `Reject Suggestions`, all words with the green go back to the wordbank 
- [x] If the user clicks `Refresh Suggestions`, MAP will recalculate. Any word that is not blue can be moved, any word that is blue will not move during "Refresh"
- [ ] If the user clicks `Refresh Suggestions` and there are no new suggestions, add a toast that says "No new suggestions"

![image](https://user-images.githubusercontent.com/13352518/39370148-d2445b88-4a03-11e8-89c4-cc328fb7a0b8.png)

![image](https://user-images.githubusercontent.com/13352518/39369579-4be45ecc-4a02-11e8-838b-fcad7cca5ac2.png)

Via Wiki: https://github.com/unfoldingWord-dev/translationCore/wiki/Alignment-Tool-Plan

## [79] TACT/MAP
This is the work that needs to happen for TACT/MAP to be usable in tC.
### [39] Preparation 
User Interface can be developed concurrently to TACT/MAP Preparation.
- [13] Publish TACT/MAP as an npm package.
- [8] Review TACT/MAP for this use case.
- [13] Address TACT/MAP issues for this use case.
###  Integration
TACT/MAP Integration comes after User Interface is developed.
- [13] Integrate TACT/MAP to predict alignment.
- [21] Address additional TACT/MAP issues that come up in integration. 
- [8] Export alignment data once complete to use for other projects.
- [8] Import alignment data from previous projects.

**Comment by da1nerd** [comment]
Note: The first step will be to implement saved alignments. Afterwards we'll continue developing the more advanced features.

**Comment by da1nerd** [comment]
When predictions are made and displayed in the wordAlignment do we want those predictions to be automatically saved or should we wait until the user clicks on them (turning them light blue) before saving them?

If we wait to save until the user clicks/drags them I can complete this without changing how the alignment data is structured in the saved files. However, if we want predicted alignments to persist to the file without user confirmation I'll need to make some drastic changes to how the alignment data is stored and managed. This option will take quite a bit longer.

Neither of these choices affect the UI designs above. This is strictly internal data management.

From the user's perspective:

1. **(click to confirm)** they will not have any unconfirmed predicted alignments in any exported data.
2. **(automatically save)** predicted alignment data will always be available in exported data.

**Comment by jag3773** [comment]
The original idea was to save the user from having to click on everything that was predicted to confirm it -- e.g. the predictions are presumed correct unless changed.  Over to @benjore on what to do.

**Comment by klappy** [comment]
We have to be careful not to save raw computer output as human aligned. That would wreak havoc on many levels. At a minimum we need to have a special attribute in the object that signifies it as Machine Generated and not human verified. This would be good to include even in the USFM 3 spec for alignment as we are about to be receiving that as import data for the Hindi.

**Comment by da1nerd** [comment]
This is what I suspected. @klappy If we simply introduce a new key to the objects we shouldn't need to worry about any migration of old project data. That would be superb!

### Here's something else I was questioning.
When/how often do we want the predictions to be made? Here are a few options:

1. Any time the user makes a change to the alignments (could be frustrating for the user?)
1. **When the user clicks on any verse.**
1. Once when the wA tool is opened.
1. Once per verse forever (flag in the project data)

While thinking about these we also need to consider what happens if a user removes a predicted alignment. Should we blacklist that alignment forever or just until the next prediction occurs?

Also, will we allow users to manually trigger new suggestions?

**Comment by da1nerd** [comment]
If the user moves to a new verse makes some changes and moves back it's possible the suggested alignments will have changed. How will we handle that? We could potentially just not overwrite them the second time but indicate that a new suggestion is available.

**Comment by benjore** [comment]
Update the alignment when the user clicks on ANY verse in the side menu (including the verse they are on). 
Add a button to `Accept all`
Add a button to `Reset all`
Add a button to `Update suggestions` (See new suggestions)

**Comment by da1nerd** [comment]
When users accept the secondary word alignments it will automatically accept the source words as merged.

**Comment by cckozie** [comment]
1. The last item in the list of check boxes in the original comment (the toast message) has not been implemented.
2. The (i) icon and buttons are left aligned in the mockups and center aligned in the implementation. (@benjore has OK'd this as implemented)
3. ~~The text on the popup when the (i) is clicked shows '(x)' rather than the circled X icon as in the mockups.~~ Moved to #2258

**Comment by da1nerd** [comment]
@cckozie regarding point thee above, we cannot add icons to localized text yet.
There is an update to the locale library we are using that supports this feature, however the library has changed enough that it will not be trivial to update. I suggest opening a new issue to tackle updating the locale library to the latest version.

**Comment by da1nerd** [comment]
I've fixed the toast display in https://github.com/translationCoreApps/wordAlignment/pull/54

**Comment by cckozie** [comment]
0.10.1 (4a8ea79)
The displaying of the toast is inconsistent. Also, the wording is different than in the description of the requirement. ('No suggestions found' vs 'No new suggestions')

[en_ult_1th_book 2.zip](https://github.com/unfoldingWord-dev/translationCore/files/2180600/en_ult_1th_book.2.zip)
In this project it appears to only be displayed when there is not text for the verse. (e.g. 1:2, 1:3)

[en_ult_1ti_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/2180594/en_ult_1ti_book.zip)
In this project it appears to only be displayed only on verses with no suggestions (e.g. not on 1:2)

**Comment by da1nerd** [comment]
@cckozie the above PR hasn't been merged in yet. Please wait for that before testing the toasts.

**Comment by mannycolon** [comment]
Waiting on `e6997db` build to be ready before moving to `In QA`

**Comment by cckozie** [comment]
I'm still seeing the same issues in 0.10.1 (88b725c)  as I was seeing above in 0.10.1 (4a8ea79)

**Comment by cckozie** [comment]
Works in 0.10.1 (b411d84)
